### PR TITLE
Rename a few factory functions in dyno

### PR DESCRIPTION
**Pull request by mppf**
Based on discussion among those of us working with dyno, we would like to use `createFor` at the beginning of factory functions that need a special name; e.g. `Resolver::createForFunction`. This PR makes that change and also removes a few `build` functions that have a better name now.

Reviewed by @DanilaFe and @arezaii - thanks!

- [x] full local testing

### CODE REVIEW

File: compiler/dyno/lib/resolution/Resolver.cpp [hunk 1]
```diff
@@ -102,19 +102,20 @@ struct GatherFieldsOrFormals {
...
   ret.curStmt = modStmt;
   ret.byPostorder.setupForSymbol(mod);
   return ret;
 }
 
 Resolver
-Resolver::moduleStmtScopeResolver(Context* context, const Module* mod,
-                                  const AstNode* modStmt,
-                                  ResolutionResultByPostorderID& byId) {
+Resolver::createForScopeResolvingModuleStmt(
```
Comment (DanilaFe) [inline compiler/dyno/lib/resolution/Resolver.cpp:115]: Now this is a mouthful...

File: compiler/dyno/test/uast/testBuildIDs.cpp [hunk 2]
```diff
@@ -147,7 +147,7 @@ static  void test2() {
   Context context;
   Context* ctx = &context;
 
-  auto builder = Builder::build(ctx, "path/to/test.chpl");
+  auto builder = Builder::createForTopLevelModule(ctx, "path/to/test.chpl");
   Builder* b   = builder.get();
```
Comment (arezaii) [inline compiler/dyno/test/uast/testBuildIDs.cpp:151]: I noticed for the parser pattern you changed from `parser.get()` to just `&parser`. Is there a reason not to do that same thing with the builder?

**Review by arezaii — APPROVED** [review]
Looks good to me, I just had a style question in the tests for builder which is more for my knowledge than something to address now.

File: compiler/dyno/test/uast/testBuildIDs.cpp (hunk 2 above)
Comment (mppf) [reply compiler/dyno/test/uast/testBuildIDs.cpp:151]: The Parser object is small, but the Builder object is not small. I was thinking there might be some advantage to heap allocating it.

**Pull request merged**